### PR TITLE
dev-java/*: Rename dev-java/{jakarta-annotations-api → jakarta-annotation-api}

### PR DESCRIPTION
**Pull request by vaukai**
---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

File: www-servers/tomcat/tomcat-10.1.31.ebuild [hunk 1]
```diff
@@ -9,7 +9,7 @@ inherit java-pkg-2 prefix verify-sig
 
 MY_P="apache-${P}-src"
 
-DESCRIPTION="Tomcat Servlet-6.0/JSP-3.1/EL-5.0/WebSocket-2.1/JASPIC-3.0 Container"
+DESCRIPTION="Servlet-6.0/JSP-3.1/EL-5.0/WebSocket-2.1/JASPIC-3.0 Container"
```
Comment (fordfrog) [inline www-servers/tomcat/tomcat-10.1.31.ebuild:12]: i'm not sure about these changes. maybe it would be better to state it similarly as on their homepage: `Apache Tomcat 10.1 is an implementation of Servlet-6.0/JSP-3.1/EL-5.0/WebSocket-2.1/JASPIC-3.0`. or if we use the same descriptions for all the ebuilds, then like this: `Apache Tomcat is an implementation of the Jakarta Servlet, Jakarta Pages, Jakarta Expression Language, Jakarta WebSocket, Jakarta Annotations and Jakarta Authentication specifications.` - but probably using abbreviations of the specifications where possible to be within the limit of description length.

so, to sum it up, we either use a generic description without the versions, or (i think) we need to use the version in te description as only that major version implements the specified specification versions.

**Review by fordfrog — CHANGES_REQUESTED** [review]

**Comment by fordfrog** [comment]
i merged the `jakarta-servlet-api` commit.

**Comment by fordfrog** [comment]
in fact, i think that not using `move` for package renames of java packages should be stated in the documentation, if it is not already there. because the system thinks after the package move that the package is already installed, but with just the package name change and without the package files being moved to a new directory, our eclasses don't work, because the location of the jar files is searched in the new directory but the files are still untouched in the old directory.

File: profiles/updates/4Q-2024 [hunk 2]
```diff
@@ -4,3 +4,4 @@ slotmove media-sound/kmetronome 5 0
 move net-im/sendxmpp-amm net-im/sendxmpp-scala
 move dev-python/jinja dev-python/jinja2
 move dev-python/gmpy dev-python/gmpy2
+move dev-java/jakarta-annotations-api dev-java/jakarta-annotation-api
```
[the remark this answers is not in the document]
Comment (vaukai) [reply profiles/updates/4Q-2024:7]: > i just tried this commit and i think this change in fact causes breakage in the tree:
> 
> ```
> # PORTDIR=/usr/src/gentoo.git/ FEATURES=test emerge -va1t cdi-api memoryfilesystem
> 
> Performing Global Updates
> (Could take a couple of minutes if you have a lot of binary packages.)
>   .='update pass'  *='binary update'  #='/var/db update'  @='/var/db move'
>   s='/var/db SLOT move'  %='binary move'  S='binary SLOT move'
>   p='update /etc/portage/package.*'
> /usr/src/gentoo.git/profiles/updates/4Q-2024.......
> @##
> 
> 
> These are the packages that would be merged, in reverse order:
> 
> Calculating dependencies... done!
> Dependency resolution took 71.59 s (backtrack: 0/20).
> 
> [ebuild  N     ] dev-java/memoryfilesystem-2.3.0-r1::gentoo  USE="-doc -source" 125 KiB
> [ebuild     U  ] dev-java/cdi-api-4.0.1-r4::gentoo [4.0.1-r3::gentoo] USE="test* -doc -source" 0 KiB
> ```

Whatever I tried, I cannot reproduce that error. With `PORTDIR=/var/db/repos/gentoo` (Yours doesn't work on my systems) it always pulls the new `dev-java/jakarta-annotation-api` as another `[ebuild  N     ]`.
And that's independent of having changed `profiles/updates/4Q-2024` or not, so I think it might be safe to omit that change. But for the documentation, I'm not sure.

**Comment by vaukai** [comment]
> in fact, i think that not using `move` for package renames of java packages should be stated in the documentation, if it is not already there. because the system thinks after the package move that the package is already installed, but with just the package name change and without the package files being moved to a new directory, our eclasses don't work, because the location of the jar files is searched in the new directory but the files are still untouched in the old directory.

Just dropped the change on `profiles/updates/4Q-2024`.

File: www-servers/tomcat/tomcat-10.1.31.ebuild (hunk 1 above)
Comment (vaukai) [reply www-servers/tomcat/tomcat-10.1.31.ebuild]: > i'm not sure about these changes. 

Indeed, if we want to do it correctly it becomes difficult. Let's keep it as is. I will drop that commit.
The more interesting question for me is: Should we try to unbundle `jakarta-annotation-api` and `jakarta-servlet-api` and possibly more?

Comment (fordfrog) [reply www-servers/tomcat/tomcat-10.1.31.ebuild]: that should be imo doable as that's the standard api and should be replaceable.

Comment (vaukai) [reply www-servers/tomcat/tomcat-10.1.31.ebuild]: > that should be imo doable as that's the standard api and should be replaceable.

Will see if it's possible without too complex patching.

**Comment by vaukai** [comment]
Keeping version bump for later when `--generate-module-info` is in the tree.

**Comment by vaukai** [comment]
This is one of the Jakarta packages to get unbundled from `www-servers/tomcat`
